hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterClientProtocol.java

```diff
@@ -2188,11 +2188,12 @@ private static FsPermission getParentPermission(final FsPermission mask) {
    * @param name Name of the mount point.
    * @param childrenNum Number of children.
    * @param date Map with the dates.
+   * @param setPath if true should set path in HdfsFileStatus
    * @return New HDFS file status representing a mount point.
    */
   @VisibleForTesting
   HdfsFileStatus getMountPointStatus(
-      String name, int childrenNum, long date) {
+      String name, int childrenNum, long date, boolean setPath) {
```

Comment:
We should leave false or true as default.

Reply:
I don't understand how to implements it without add a boolean argument in getMountPointStatus().

Reply:
I'm just saying that you can keep the old method and call whicheves is the default value:
```
HdfsFileStatus getMountPointStatus(
    String name, int childrenNum, long date) {
  return getMountPointStatus(name, childrenNum, date);
}
```
Then you don't need to modify that many other pieces of code.